energy-growers/src/components/AccordionList/Accordion/Accordion.tsx

```diff
@@ -0,0 +1,46 @@
+"use client";
+import { useEffect, useRef, useState } from "react";
+import "./Accordion.scss";
+import Icon from "../../Icon/Icon";
+import { AccordionItem } from "@/models/accordion";
+
+const Accordion: React.FC<AccordionItem> = ({
+  data,
+  isOpen = false,
+  btnOnClick,
+}) => {
+  const [height, setHeight] = useState<number | undefined>(0);
+  const ref = useRef<any | null>(null);
+
+  useEffect(() => {
+    setHeight(isOpen ? ref.current?.scrollHeight : 0);
+  }, [isOpen]);
+
+  return (
+    <div className="eg-container-accordion eg-description">
+      <button className="eg-content-accordion" onClick={btnOnClick}>
+        <span className={`eg-question-accordion  ${isOpen ? "eg-open-accordion" : ""}`}>
+          {data?.question}
+        </span>
+        <Icon
+          type="arrow"
+          iconOnly
+          iconType="general"
+          noHover
+          className={`${isOpen ? "eg-open-accordion eg-rotate-arrow" : ""}`}
+        />
+      </button>
+      <div
+        ref={ref}
+        className={`eg-accordion-item-container ${
+          isOpen ? "eg-open-accordion-answer" : ""
+        }`}
```

Comment:
```suggestion
        className={`eg-accordion-item-container ${getClasses({
          "eg-open-accordion-answer": isOpen,
        })}`}
```

utilizziamo le cose fatte, non le usiamo solo quando le facciamo